### DISCUSSION TITLE
is "vulnerabilityAlerts" for supported self-hosted renovate on gitlab platform

### DISCUSSION THREAD

**Discussion by prasadthotakura** [renovatebot/renovate#15561] [Request Help] [answered]
I'm trying to use self-hosted renovate(v32.39) that can run part of our gitlab(v14.7.7 enterprise) CI pipeline job for updating ONLY the open source libraries security vulnerabilities. I'm trying following rules in config.json

```
"packageRules": [  
    {  
      "matchDatasources": ["gradle-version", "maven"],  
      "registryUrls": [  
        "registry1",  
        "registry2",  
        "registry3"  
      ] 
    },
    {
      "matchPackageNames": ["*"],
      "matchUpdateTypes": ["major", "minor", "patch"]
    }
  ],
"vulnerabilityAlerts": {
    "enabled": true,
    "labels": ["security bot updates"]
  }
```
And I see the following debug statement in the logs

`DEBUG: No vulnerability alerts found (repository=samplerepo)`

I expect MRs to be created only for libraries that have security vulnerabilities. But I see renovate is creating the MRs for libraries that don't have any security vulnerabilities.

is `vulnerabilityAlerts` supposed to work for gitlab?

**Comment by rarkins** [comment]
No, it uses GitHub's alerts API for each repo

**Reply by prasadthotakura** [reply]
It would be great to have similar support for gitlab as well. 
Is there any way that we can  extend/override to call gitlab APIs instead of github APIs?

**Reply by viceice** [reply]
no, we working on an alternative, but that needs time.

**Reply by rarkins** [reply]
> It would be great to have similar support for gitlab as well. 
> 
> Is there any way that we can  extend/override to call gitlab APIs instead of github APIs?

Which gitlab API?

**Reply by prasadthotakura** [reply]
vulnerability_findings API

**Reply by rarkins** [reply]
Looks to be an ultimate only feature, which is a very small percentage of our gitlab user base.

**Comment by philippe-granet** [comment, marked as answer]
@prasadthotakura Release https://github.com/renovatebot/renovate/releases/tag/34.129.0 add feature-flagged support for OSV (https://github.com/renovatebot/renovate/issues/20226) ([a91ca62](https://github.com/renovatebot/renovate/commit/a91ca62b466b57aa9a18f4e15bd04eb5552278b1))

https://docs.renovatebot.com/configuration-options/#osvvulnerabilityalerts
> Renovate integrates with [OSV](https://osv.dev/), an open-source vulnerability database, to check if extracted dependencies have known vulnerabilities.
> Set osvVulnerabilityAlerts to true to get pull requests with vulnerability fixes (once they are available).
